src/http-operation.ts

```diff
@@ -0,0 +1,199 @@
+import { INode, INodeExample } from './graph';
+import { ISchema } from './schemas';
+import { IServer } from './servers';
+
+export interface IHttpOperation extends INode {
+  method: string;
+  path: string;
+  responses: IHttpOpResponse[];
+  servers?: IServer[];
+  request?: IHttpOpRequestParams;
+  security?: HttpOpSecurityScheme[];
+  deprecated?: boolean;
+}
+
+// Inspired by: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameterObject
+export interface IHttpOpParam {
+  name: string;
+  style?: HttpOpParamStyles;
+  description?: string;
+  required?: boolean;
+  explode?: boolean;
+  deprecated?: boolean;
+
+  // most of the type, params from oas2/oas3 will simply be converted to content with one '*' key
+  content: {
+    [mediaType: string]: IHttpOpContent;
+  };
+}
+
+export const enum HttpOpParamStyles {
+  Simple = 'simple',
+  Matrix = 'matrix',
+  Label = 'label',
+  Form = 'form',
+  SpaceDelimited = 'spaceDelimited',
+  PipeDelimited = 'pipeDelimited',
+  DeepObject = 'deepObject',
+}
+
+export interface IHttpOpPathParam extends IHttpOpParam {
+  // defaults to simple
+  style?: HttpOpParamStyles.Label | HttpOpParamStyles.Matrix | HttpOpParamStyles.Simple;
+}
+
+export interface IHttpOpQueryParam extends IHttpOpParam {
+  // defaults to form
+  style?:
+    | HttpOpParamStyles.Form
+    | HttpOpParamStyles.SpaceDelimited
+    | HttpOpParamStyles.PipeDelimited
+    | HttpOpParamStyles.DeepObject;
+
+  allowEmptyValue?: boolean;
```

Comment:
According to Aristoteles, _tertium non datur_ — so either something is true or false. Why can this be `null`?

Reply:
It can't be null, it can be undefined :). Basically, you do not have to set the property every time you want to define an object w this interface. However, we might want to just make it required for clarity, @chris-miaskowski?

Reply:
In that case my question is — what would am implementation using such interface do when this property is not defined? Would it allow an empty value, or would it not?

Reply:
Would allow undefined value and basically treat it as false.

```
if (param.allowEmptyValue) {
  // do whatever
}
```

Leaving it as optional is convenient so that one does not always have to initialize all of these boolean props to false everywhere.